sw/device/silicon_creator/mask_rom/mask_rom_start.S

```diff
@@ -148,17 +149,24 @@ _mask_rom_start_boot:
   // The following sequence enables the minimum level of entropy required to
   // initialize memory scrambling, as well as the entropy distribution network.
   // FIXME: Enable entropy complex - this is not the full enable.
-  li   a0, TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR
-  li   t0, 0x2
-  sw   t0, ENTROPY_SRC_CONF_REG_OFFSET(a0)
+  // TODO(#7221): Switch entropy source mode from LFSR to PTRNG.
+  li a0, TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR
+  li t0, (2 << ENTROPY_SRC_CONF_ENABLE_OFFSET) // LFSR mode.
+  sw t0, ENTROPY_SRC_CONF_REG_OFFSET(a0)
 
-  li   a0, TOP_EARLGREY_CSRNG_BASE_ADDR
-  li   t0, 0x1
-  sw   t0, CSRNG_CTRL_REG_OFFSET(a0)
+  li a0, TOP_EARLGREY_CSRNG_BASE_ADDR
+  li t0, (1 << CSRNG_CTRL_ENABLE_BIT)
+  sw t0, CSRNG_CTRL_REG_OFFSET(a0)
 
-  li   a0, TOP_EARLGREY_EDN0_BASE_ADDR
-  li   t0, 0x9
-  sw   t0, EDN_CTRL_REG_OFFSET(a0)
+  li a0, TOP_EARLGREY_EDN0_BASE_ADDR
+  li t0, (1 << EDN_CTRL_EDN_ENABLE_BIT) | (EDN_CTRL_HW_REQ_MODE_VALUE_BOOT_REQ_MODE << EDN_CTRL_HW_REQ_MODE_OFFSET)
+  sw t0, EDN_CTRL_REG_OFFSET(a0)
+
+  // Scramble and initialize main memory (main SRAM).
+  // Memory accesses will stall until initialization is complete.
+  li a0, TOP_EARLGREY_SRAM_CTRL_MAIN_BASE_ADDR
+  li a1, (1 << SRAM_CTRL_CTRL_RENEW_SCR_KEY_BIT) | (1 << SRAM_CTRL_CTRL_INIT_BIT)
```

Comment:
@msfschaffner 
one thing that i realized about the memory init... because the integrity right now is handled inside `tlul_adapter_sram`, the sram init actually still does not produce correct integrity.  

This is probably something we'll want to update and allow the init to happen either in the sram itself or from the tlul adapter sram.

Reply:
yes I can update that. I am in the process of absorbing the ram primitive into the sram_ctrl, so I can technically pull the LFSR back out into the sram_ctrl and compute the correct ECC. let me make an issue for this.

Reply:
do you think it makes sense to give the tlul_adapter_sram the "init" ability?
The ecc encoders are tiny..so it doesn't bother me too much, but it feels a little weird to have duplicate copies in tlul_adapter_sram (for the byte purpose), and also one in sram_ctrl.

let me know what you think.

Reply:
we can also move it into `tlul_adapter_sram`, yes. maybe this init mechanism comes in handy in some other designs in the future. we just have to be aware of the coverage impact in existing modules that make use of this module...

Reply:
Forgot to link the issue here:
https://github.com/lowRISC/opentitan/issues/7225